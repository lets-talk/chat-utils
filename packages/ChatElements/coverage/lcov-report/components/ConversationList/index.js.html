<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ConversationList/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ConversationList</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/8</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MdChatBubble from 'react-icons/lib/md/chat-bubble';
&nbsp;
import { withAutoScroll } from '../../utils/hoc';
&nbsp;
import ConversationBox from '../ConversationBox';
import './index.scss';
&nbsp;
class ConversationList extends Component {
<span class="fstat-no" title="function not covered" >  on</span>Click(item, i, e) {
<span class="cstat-no" title="statement not covered" >    if (this.props.clickItem instanceof Function) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.clickItem(item, i, e);</span>
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  re</span>nder() {
<span class="cstat-no" title="statement not covered" >    return (</span>
      &lt;div ref={this.props.cmpRef} id="conversations" className="letstalk-conversations-container"&gt;
        {this.props.conversations.length &gt; 0 &amp;&amp;
          &lt;div className="letstalk-conversations-list"&gt;
            {this.props.conversations.map(<span class="fstat-no" title="function not covered" >(c</span>onversation, index) =&gt;
              (
<span class="cstat-no" title="statement not covered" >                &lt;div className="letstalk-conversation-container" key={index}&gt;</span>
                  &lt;ConversationBox
                    conversation={conversation}
                    key={`conversation-${index}`}
                    onClick={this.props.clickItem &amp;&amp; (<span class="fstat-no" title="function not covered" >(e</span>) =&gt; <span class="cstat-no" title="statement not covered" >this.onClick(conversation, index, e))</span>}
                  /&gt;
                &lt;/div&gt;
              ))
            }
            &lt;div className="conversations-list-separator"&gt;{this.props.noMoreDataText}&lt;/div&gt;
          &lt;/div&gt;
        }
&nbsp;
        {this.props.conversations.length === 0 &amp;&amp;
          &lt;div className="letstalk-conversations-empty-state-container"&gt;
            &lt;div className="svg-container"&gt;
              &lt;MdChatBubble color="#e5e9ec" size={40} /&gt;
            &lt;/div&gt;
            &lt;div className="empty-state"&gt;
              {this.props.emptyStateText}
            &lt;/div&gt;
          &lt;/div&gt;
        }
      &lt;/div&gt;
    );
  }
}
&nbsp;
ConversationList.propTypes = {
  /**
   * conversations is an array of conversations objects
   * that will be displayed in the component
   */
  conversations: PropTypes.array,
  /**
   * Callback function to be called when an item is clicked
   */
  clickItem: PropTypes.func,
  /**
   * Text to be used when no conversations
   */
  emptyStateText: PropTypes.string,
  /**
   * cmpRef is a function that is called by anyone
   * that needs an instance ref of this component
   */
  cmpRef: PropTypes.func,
};
&nbsp;
ConversationList.defaultProps = {
  conversations: [],
  clickItem: null,
  cmpRef: null,
  emptyStateText: '',
};
&nbsp;
const autoScrollOptions = { threshold: 300, direction: 'top' };
export default withAutoScroll(autoScrollOptions)(ConversationList);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
