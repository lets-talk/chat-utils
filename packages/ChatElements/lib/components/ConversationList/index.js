import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MdChatBubble from 'react-icons/lib/md/chat-bubble';

import { withAutoScroll } from '../../utils/hoc';

import ConversationBox from '../ConversationBox';
import './index.scss';

class ConversationList extends Component {
  onClick(item, i, e) {
    if (this.props.clickItem instanceof Function) {
      this.props.clickItem(item, i, e);
    }
  }

  render() {
    return (
      <div ref={this.props.cmpRef} id="conversations" className="letstalk-conversations-container">
        {this.props.conversations.length > 0 &&
          <div className="letstalk-conversations-list">
            {this.props.conversations.map((conversation, index) =>
              (
                <div className="letstalk-conversation-container" key={index}>
                  <ConversationBox
                    conversation={conversation}
                    key={`conversation-${index}`}
                    onClick={this.props.clickItem && ((e) => this.onClick(conversation, index, e))}
                  />
                </div>
              ))
            }
            <div className="conversations-list-separator">{this.props.noMoreDataText}</div>
          </div>
        }

        {this.props.conversations.length === 0 &&
          <div className="letstalk-conversations-empty-state-container">
            <div className="svg-container">
              <MdChatBubble color="#e5e9ec" size={40} />
            </div>
            <div className="empty-state">
              {this.props.emptyStateText}
            </div>
          </div>
        }
      </div>
    );
  }
}

ConversationList.propTypes = {
  conversations: PropTypes.array,
  clickItem: PropTypes.func,
  emptyStateText: PropTypes.string,
  /**
   * cmpRef is a function that is called by anyone
   * that needs an instance ref of this component
   */
  cmpRef: PropTypes.func,
};

ConversationList.defaultProps = {
  conversations: [],
  clickItem: null,
  cmpRef: null,
  emptyStateText: '',
};

const autoScrollOptions = { threshold: '100%', direction: 'top' };
export default withAutoScroll(autoScrollOptions)(ConversationList);
