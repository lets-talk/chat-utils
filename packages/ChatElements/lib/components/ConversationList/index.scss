@import "./config.scss";

@mixin letstalk-conversations-container {
  height: 100%;
  max-height: none;
  font-size: $font-size-normal;
}

.letstalk-conversations-container {
  background-color: $widgetBox-MessageList-background;
  height: 50vh;
  min-height: 200px;
  max-height: 410px;
  overflow-y: scroll;
  padding: 10px;
}

.full-screen {
  .letstalk-conversations-container {
    @include letstalk-conversations-container;
  }
}

@media screen and (max-width: 800px)  {
  .letstalk-conversations-container {
    @include letstalk-conversations-container;
  }
}
