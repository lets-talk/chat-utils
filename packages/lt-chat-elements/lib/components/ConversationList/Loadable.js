import React from 'react';
import path from 'path';
import Loadable from 'react-loadable';
import LoadingComponent from '../Loading';

const sleep = (duration) => new Promise((resolve) => {
  setTimeout(() => {
    resolve();
  }, duration);
});

const LoadableConversationList = Loadable({
  loader: () => sleep(5 * 1000).then(() => import('./index')),
  loading: LoadingComponent,
  serverSideRequirePath: path.resolve(__dirname, './index'),
});

const ConversationList = (props) => (<LoadableConversationList {...props} />);

export default ConversationList;
