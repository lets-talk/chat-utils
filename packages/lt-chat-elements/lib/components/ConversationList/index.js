import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MdChatBubble from 'react-icons/lib/md/chat-bubble';
import styled, { css } from 'styled-components';

import { withAutoScroll } from '../../utils/hoc';
import { animate } from '../../utils/style';

import ConversationBox from '../ConversationBox';

const ConversationListWrapper = styled.div`
  display: flex;
  flex-direction: column;
  max-height: none;
  background-color: ${(props) => props.theme.palette.common.white};
  height: 50vh;
  min-height: 200px;
  overflow-y: auto;
  ${css`
      animation: ${animate('slide-in')} 3s;
    `};
  `;

const ConversationListSeparator = styled.div`
  background-color: ${(props) => props.theme.palette.common.white};
  font-weight: ${(props) => props.theme.typography.fontWeightThin};
  margin-top: 10px;
  margin-left: 80px;
  padding: 20px 0px;
  color: ${(props) => props.theme.palette.text.primary};
  border-top: 1px solid ${(props) => props.theme.palette.divider};
`;

const NoConversation = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center; /*centers items on the line (the x-axis by default)*/
  align-items: center; /*centers items on the cross-axis (y by default)*/
  flex-direction: column;
`;

const SvgContainer = styled.div`
  margin-bottom: 10px;
`;

const EmptyContainer = styled.div`
  max-width: 80%;
`;

class ConversationList extends Component {
  onClick(item, i, e) {
    if (this.props.clickItem instanceof Function) {
      this.props.clickItem(item, i, e);
    }
  }

  render() {
    return (
      <ConversationListWrapper
        ref={this.props.cmpRef}
        className={this.props.className}
      >
        {this.props.conversations.length > 0 &&
          <div>
            {this.props.conversations.map((conversation, index) =>
              (
                <div
                  role="button"
                  tabIndex={index + 1}
                  key={conversation.id}
                  onClick={this.props.clickItem && ((e) => this.onClick(conversation, index, e))}
                  onKeyPress={this.props.clickItem && ((e) => this.onClick(conversation, index, e))}
                >
                  <ConversationBox
                    conversation={conversation}
                    key={`conversation-${conversation.id}`}
                  />
                </div>
              ))
            }
            <ConversationListSeparator>{this.props.noMoreDataText}</ConversationListSeparator>
          </div>
        }

        {this.props.conversations.length === 0 &&
          <NoConversation>
            <SvgContainer>
              <MdChatBubble color="#e5e9ec" size={40} />
            </SvgContainer>
            <EmptyContainer>
              {this.props.emptyStateText}
            </EmptyContainer>
          </NoConversation>
        }
      </ConversationListWrapper>
    );
  }
}

ConversationList.propTypes = {
  /**
   * conversations is an array of conversations objects
   * that will be displayed in the component
   */
  conversations: PropTypes.array,
  /**
   * Callback function to be called when an item is clicked
   */
  clickItem: PropTypes.func,
  /**
   * Extra className to style the component
   */
  className: PropTypes.string,
  /**
   * Text to be used to show noMoreDataText
   */
  noMoreDataText: PropTypes.string,
  /**
   * Text to be used when no conversations
   */
  emptyStateText: PropTypes.string,
  /**
   * cmpRef is a function that is called by anyone
   * that needs an instance ref of this component
   */
  cmpRef: PropTypes.func,
};

ConversationList.defaultProps = {
  conversations: [],
  clickItem: null,
  cmpRef: null,
  emptyStateText: '',
  noMoreDataText: '',
};

const autoScrollOptions = { threshold: 300, direction: 'top' };
export default withAutoScroll(autoScrollOptions)(ConversationList);
