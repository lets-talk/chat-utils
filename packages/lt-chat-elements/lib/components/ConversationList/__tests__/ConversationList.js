import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConversationList from '../index';

describe('ConversationList component', () => {
  const conversations = [
    {
      client: {
        name: 'Georgia Harmon',
        avatar: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
      },
      last_message: {
        content: 'Lorem ipusum la la ejemplo largo',
        created_at: '2017-11-20T09:43:57.000-0300',
      },
      tags: [
        {
          name: 'SPU-CLAVE',
        },
      ],
    },
  ];

  it('should render with empty props', () => {
    const component = shallow(<ConversationList />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with custom props', () => {
    const props = {
      conversations,
      clickItem: null,
      cmpRef: null,
      emptyStateText: 'No hay mas conversaciones',
    };

    const component = shallow(<ConversationList {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with no conversation the empty state', () => {
    const props = {
      conversations: [],
      clickItem: null,
      cmpRef: null,
      emptyStateText: 'No hay mas conversaciones',
    };

    const component = shallow(<ConversationList {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('Test onClick event ', () => {
    const mockCallBack = jest.fn();
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

    const props = {
      conversations,
      cmpRef: null,
      emptyStateText: 'No hay mas conversaciones',
    };

    // Need to call mount because the click is triggered in the child ConversationBox item element
    const component = mount(<ConversationList {...props} clickItem={() => mockCallBack()} />);
    component.find('.letstalk-conversation-container').simulate('click', fakeEventObject);
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
});
