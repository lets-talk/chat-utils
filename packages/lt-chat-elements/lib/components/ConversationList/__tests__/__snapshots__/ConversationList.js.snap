// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConversationList component should render with custom props 1`] = `
<ConversationList
  clickItem={null}
  cmpRef={null}
  conversations={
    Array [
      Object {
        "client": Object {
          "avatar": "https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg",
          "name": "Georgia Harmon",
        },
        "last_message": Object {
          "content": "Lorem ipusum la la ejemplo largo",
          "created_at": "2017-11-20T09:43:57.000-0300",
        },
        "tags": Array [
          Object {
            "name": "SPU-CLAVE",
          },
        ],
      },
    ]
  }
  emptyStateText="No hay mas conversaciones"
  noMoreDataText=""
/>
`;

exports[`ConversationList component should render with empty props 1`] = `
<ConversationList
  clickItem={null}
  cmpRef={[Function]}
  conversations={Array []}
  emptyStateText=""
  noMoreDataText=""
/>
`;

exports[`ConversationList component should render with no conversation the empty state 1`] = `
<ConversationList
  clickItem={null}
  cmpRef={null}
  conversations={Array []}
  emptyStateText="No hay mas conversaciones"
  noMoreDataText=""
/>
`;
